core/tests/test_transform.h

```diff
@@ -0,0 +1,265 @@
+//
+//  test_transform.h
+//  engine-unit-tests
+//
+//  Created by Xavier Legland on 10/05/2022.
+//
+
+#pragma once
+
+#include "scene.h"
+#include "transform.h"
+
+void test_transform_rotation_position(void) {
+    // set rotation
+    {
+        Transform* t = transform_make_default();
+        TEST_ASSERT(t != NULL);
+        Quaternion* q = quaternion_new(1.0f, 2.0f, 3.0f, 0.5f, false);
+        transform_set_rotation(t, q);
+        Quaternion* r = transform_get_rotation(t);
+        TEST_CHECK(quaternion_is_equal(r, q, EPSILON_ZERO));
+        TEST_CHECK(r->normalized == q->normalized);
+
+        quaternion_free(q);
+        transform_release(t);
+    }
+
+    // set position
+    {
+        Transform* t = transform_make_default();
+        TEST_ASSERT(t != NULL);
+        transform_set_position(t, 1.0f, 2.0f, 3.0f);
+        const float3* pos1 = transform_get_position(t);
+        const float3 expected = { 1.0f, 2.0f, 3.0f };
+        TEST_CHECK(float3_isEqual(pos1, &expected, EPSILON_ZERO));
+        transform_release(t);
+    }
+
+    // set position (with a float3)
+    {
+        Transform* t = transform_make_default();
+        TEST_ASSERT(t != NULL);
+        const float3* pos1 = transform_get_position(t);
+        float3* pos2 = float3_new(4.0f, 5.0f, 6.0f);
+        const float3 expected = { 4.0f, 5.0f, 6.0f };
+        transform_set_position_vec(t, pos2);
+        TEST_CHECK(float3_isEqual(pos1, &expected, EPSILON_ZERO));
+
+        float3_free(pos2);
+        transform_release(t);
+    }
```

Comment:
not sure a malloc of float3 (`pos2`) is required here
Let's keep it simple. What do you think of this ?

```cpp
    // set position (with a float3)
    {
        Transform* t = transform_make_default();
        TEST_ASSERT(t != NULL);

        const float3 pos = { 4.0f, 5.0f, 6.0f };
        transform_set_position_vec(t, &pos);

        const float3* curPos = transform_get_position(t);
        const float3 expected = { 4.0f, 5.0f, 6.0f };
        TEST_CHECK(float3_isEqual(curPos, &expected, EPSILON_ZERO));

        transform_release(t);
    }
```